django/db/migrations/autodetector.py

```diff
@@ -1367,13 +1367,39 @@ def generate_added_constraints(self):
             app_label,
             model_name,
         ), alt_constraints in self.altered_constraints.items():
+            dependencies = []
+
             for constraint in alt_constraints["added_constraints"]:
+                model_state = self.to_state.models[app_label, model_name]
+
+                fields = list(model_state.fields.values())
+
```

Comment:
All of this code could be moved one level up to the outer-loop context as it is a function of `app_label, model_name` and not `constraint`